pkg/chart/dependency.go

```diff
@@ -204,3 +207,46 @@ func lockFilePath(chartPath, apiVersion string) (string, error) {
 		return "", errors.Errorf("unrecognised apiVersion %q", apiVersion)
 	}
 }
+
+// GetChartDependencies returns the chart dependencies from a chart in tgz format.
+func GetChartDependencies(filepath string, name string) ([]*helmChart.Dependency, error) {
+	// Create temporary working directory
+	chartPath, err := ioutil.TempDir("", "charts-syncer")
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer os.RemoveAll(chartPath)
+
+	// Uncompress chart
+	if err := utils.Untar(filepath, chartPath); err != nil {
+		return nil, errors.Annotatef(err, "uncompressing %q", filepath)
+	}
+	// Untar uncompress the chart in a subfolder
+	chartPath = path.Join(chartPath, name)
+
+	var apiVersion string
+	if _, err := os.Stat(path.Join(chartPath, RequirementsLockFilename)); err == nil {
+		apiVersion = APIV1
+	}
+	if _, err := os.Stat(path.Join(chartPath, ChartLockFilename)); err == nil {
+		apiVersion = APIV2
+	}
+
+	if apiVersion == "" {
+		return nil, nil
```

Comment:
shouldn't we return an error in this case?

Reply:
I think this is a bit unclear.

We only set `apiVersion` if any of the possible "lock" files are found. There are charts without a "lock" file so this `if` clause is covering that case. Should I add a comment?

Reply:
I see...I thought you were only calling this function IF the chart has dependencies but it's been called for all charts. In that case yes, it makes sense that we return nil with no error.

What I find wrong is that we are using apiVersion for that. In order to check the API, we should be checking the Chart.yaml file and additionally check if there are deps or not.

Reply:
I see your point. However, that also depends on the API version. I guess you were doing this quick check to identify the API version right? I assume all charts are well-formed so we don't need to validate if we miss the dependencies in the "main" file.

Reply:
Yes, it depends on the API version, but we don't know if charts are really well-formed or not. Maybe someone in one repo has pushed a V1 chart with a Chart.lock file (which is from v2). In that case, we would think the chart is V2. Although probably that wrong chart will fail later because of another reason.

In any case, I was just stating that as a future improvement, I know it was already like that.

Reply:
I think it is not the responsibility of this tool to manage that case. If that ever happens, any tool consuming that chart would fail, not only this syncer.